File: leetcode/7.cpp
#include <climits>

class Solution {
public:
  int reverse(int x) {
    if (x < 10 && x > -10) {
      return x;
    }
    bool dir = x > 0;
    x = x > 0 ? x : -x;
    long ans = 0;
    while (x > 0) {
      ans = ans * 10 + x % 10;
      x /= 10;
    }
    ans = ans * (dir ? 1 : -1);
    if (ans < INT_MIN || ans > INT_MAX) {
      return 0;
    }
    return ans;
  }
};

File: leetcode/7.py
class Solution:

    def reverse(self, x: int) -> int:
        ans = 0
        x, dir = (x, 1) if x >= 0 else (-x, -1)
        while x > 0:
            ans = ans * 10
            ans += x % 10
            x = x // 10
        ans = dir * ans
        return ans if ans <= 2147483647 and ans >= -2147483648 else 0
